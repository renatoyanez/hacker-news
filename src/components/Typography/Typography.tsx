import { ReactNode } from 'react';
import styles from './typography.module.css';

interface TypographyProps {
	children?: ReactNode | ReactNode[];
	type?: string;
}

const Typography = ({
	type = 'textStyle',
	children,
}: TypographyProps) => {
	return (
		<>
			<p className={styles[type]}>{children}</p>
		</>
	);
};

export default Typography;
