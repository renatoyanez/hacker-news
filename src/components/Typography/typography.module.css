@import '../../styles/variables.css';

.textStyle {
	font-family: Baskerville;
	font-size: 1.75rem;
	font-weight: normal;
	font-stretch: normal;
	font-style: normal;
	line-height: 1;
	letter-spacing: normal;
	color: #3b3b3b;
	margin: 0;
}
.textStyle2 {
	font-family: Roboto;
	font-size: 1rem;
	font-weight: 500;
	font-stretch: normal;
	font-style: normal;
	line-height: 1.75;
	letter-spacing: normal;
	text-align: center;
	color: var(--azure);
	margin: 0;
}
.textStyle4 {
	font-family: Roboto;
	font-size: 0.875rem;
	font-weight: 500;
	font-stretch: normal;
	font-style: normal;
	line-height: 1.43;
	letter-spacing: 0.25px;
	color: #606060;
	margin: 0;
}
.textStyle3 {
	font-family: Roboto;
	font-size: 0.875rem;
	font-weight: 500;
	font-stretch: normal;
	font-style: normal;
	line-height: 1.43;
	letter-spacing: 0.25px;
	color: var(--brownish-grey);
	margin: 0;
}

.textStyle5 {
	font-family: Roboto;
	font-size: 1rem;
	font-weight: 500;
	font-stretch: normal;
	font-style: normal;
	line-height: 1.75;
	letter-spacing: normal;
	text-align: center;
	color: #606060;
	margin: 0;
}

.textStyle6 {
	font-family: Roboto;
	font-size: 0.688rem;
	font-weight: normal;
	font-stretch: normal;
	font-style: normal;
	line-height: normal;
	letter-spacing: normal;
	color: #767676;
	margin: 0;
}

.textStyle7 {
	font-family: Roboto;
	font-size: 0.875rem;
	font-weight: 500;
	font-stretch: normal;
	font-style: normal;
	line-height: 1.43;
	letter-spacing: 0.25px;
	color: white;
	margin: 0;
}
