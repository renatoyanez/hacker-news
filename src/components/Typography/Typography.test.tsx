import React from 'react';
import { create } from 'react-test-renderer';
import Typography from './Typography';

test('Typography renders correctly', () => {
	const tree = create(
		<Typography type="textStyle2">test</Typography>,
	).toJSON();
	expect(tree).toMatchSnapshot();
});
