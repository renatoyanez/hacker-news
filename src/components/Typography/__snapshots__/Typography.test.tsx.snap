// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders correctly 1`] = `
<p
  className="textStyle2"
>
  test
</p>
`;
